### PR TITLE
auth: ratelimit logins

### PR DESCRIPTION
**Pull request by mattrobenolt**
🤕

### CODE REVIEW

**Comment by mitsuhiko** [comment]
I know it says it's good enough for anybody but theoretically this might turn into an issue for large corps that all have people sign in at 8 in the morning.

**Review by mitsuhiko — APPROVED** [review]

**Comment by mitsuhiko** [comment]
Probably not a huge deal though.

**Comment by mattrobenolt** [comment]
> theoretically this might turn into an issue for large corps that all have people sign in at 8 in the morning.

Yeah, it think the odds of it all happening literally within the same minute would be rough.

Alternative, it could rate limit by the username attempting to login as. To avoid per-ip rate limit, but you can't brute force into a single account.

On second thought, this might be better anyways to block a brute force attempt across many IPs to attack one single account.

**Comment by mitsuhiko** [comment]
Yeah, I think that might be a better idea. And then you can still do an IP based rate limit with some larger limit.

File: src/sentry/web/frontend/auth_login.py [hunk 1]
```diff
@@ -87,15 +87,26 @@ def handle_basic_auth(self, request):
...
+            from sentry.utils.hashlib import md5_text
 
-            auth.login(request, user)
+            login_attempt = op == 'login' and request.POST.get('username') and request.POST.get('password')
 
-            if not user.is_active:
-                return self.redirect(reverse('sentry-reactivate-account'))
+            if login_attempt and ratelimiter.is_limited(
+                u'auth:login:username:{}'.format(md5_text(request.POST['username'].lower()).hexdigest()),
+                limit=2, window=60,  # 10 per minute should be enough for anyone
```
Comment (mitsuhiko) [inline src/sentry/web/frontend/auth_login.py]: the comment does not match the limit

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**